crates/bevy_render/src/render_graph/nodes/render_resources_node.rs

```diff
@@ -434,6 +436,14 @@ fn render_resources_node_system<T: RenderResources>(
             .uniform_buffer_arrays
             .increment_changed_item_counts(&uniforms);
     }
+
+    // update uniforms info
+    for (uniforms, _dispatch, _compute_pipelines) in &mut query2.iter() {
```

Comment:
In its current state, I don't think the (non-asset) RenderResourcesNode will behave correctly for compute:

1. it never reads buffer data from the compute query
2. if an entity has both ComputePipelines and RenderPipelines, we will double increment changed item count.

Lets either fix these problems or just remove the code and add an issue to follow up

Reply:
>     1. it never reads buffer data from the compute query
I'm not sure I understand what you mean here?


>     2. if an entity has both ComputePipelines and RenderPipelines, we will double increment changed item count.
Yeah, I think we need more separation between them. Ideally compute is never run on a single component as that would just be wasteful. I don't think we want to encourage people to have large components.. Instead we really need to work the compute API as it stands now to be system based and process multiple components. An example:
```rust
// Compute system pseudo code..
fn compute_velocity(
  compute_dispatcher: ResMut<ComputeDispatcher>,
  velocity_pipeline: ResMut<VelocityPipeline>,
  velocity_compute_resource: ResMut<VelocityComputeResource>
  query: Query<(&Translation, &Velocity)>,
) {

  // Imagine this a bit more verbose..
  query.translations = velocity_compute_resource.read();

  // Imagine this a bit more verbose..
  velocity_compute_resource.write(query);

  compute_dispatcher.set_pipeline(velocity_pipeline);

  // Adds a resource to be used by the compute shader for the current frame.
  compute_dispatcher.add_resource(velocity_compute_resource)

  // Add dispatch command for the current frame.
  // Again imagine something more verbose here..
  compute_dispatcher.dispatch(query.len());
}
```

Reply:
Yeah I agree that compute should generally be run on sets of components. I imagine there will be a number of cases where people will want to run compute on "all transforms" or "all material assets" as input. It probably makes sense to make it easy to reuse the buffers generated by RenderResourcesNode (and ditto for AssetRenderResourcesNode). They are already contiguous arrays of data (although each value in the array is aligned to 256 bits in order to meet the "dynamic uniform" constraints). Reusing these abstractions also makes it "easy" to operate on ECS data.

> I'm not sure I understand what you mean here?

Currently you call "increment_changed_item_counts" which sets up BufferArrayStatus to track the render resources that are returned from the "compute_query". But then you don't actually copy buffer data. The only code that currently does that is this: https://github.com/bevyengine/bevy/blob/0958427f7c5abd534c757c9e3112df3f6d7c358b/crates/bevy_render/src/render_graph/nodes/render_resources_node.rs#L472

But that only runs on the "draw query" not the "compute query"